platform/checkout/template.j2

```diff
@@ -0,0 +1,6 @@
+{% set path = env('PLATFORM_PATH') %}
+{% if module.ref is defined %}
```

Comment:
@xumia 
This change is making `ref` optional. Would this meet the azp requirement?
This was added after we had shared the diff last week.

Reply:
@huanlev , Azp can do the 'git clone {{ module.repo }} {{ path }} '.
How about add a new step to do the clone, if {{ path }} (or {{ path }}/.git ) is existing, then skip the "git clone"

Reply:
@huanlev, you mean the build target folder has already existed, then it will not do the step, right?
Azp can only clone the static branch/commit id, it can not be dynamic. We need to do checkout the right commit.

Reply:
Got it. We understand the use-case and would implement a solution for this @xumia

Reply:
diffs updated. Please review.